File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/java/tests/ThirdScreenTest.java
package pavelnazimok.uitestingfeatures.java.tests;

import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.Random;

@RunWith(AndroidJUnit4.class)
public class ThirdScreenTest extends BaseTest {

    @Test
    public void itemText_afterSelectInListView_shouldBeDisplayed() {
        menuRobot
                .clickThirdScreen();

        final String selectedItemText = thirdScreenRobot
                .clickShowListView()
                .clickListViewItem(new Random().nextInt(thirdScreenRobot.getListViewCount()));

        thirdScreenRobot
                .assertSelectedItemIsDisplayed(selectedItemText);
    }
}

File: app/src/androidTest/java/pavelnazimok/uitestingfeatures/kotlin/tests/ThirdScreenTest.kt
package pavelnazimok.uitestingfeatures.kotlin.tests

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Test
import org.junit.runner.RunWith
import pavelnazimok.uitestingfeatures.kotlin.robots.onMenu
import pavelnazimok.uitestingfeatures.kotlin.robots.onThirdScreen
import kotlin.random.Random

@RunWith(AndroidJUnit4::class)
class ThirdScreenTest : BaseTest() {

    @Test
    fun itemText_afterSelectInListView_shouldBeDisplayed() {
        onMenu {
            clickThirdScreen()
        }

        onThirdScreen {
            clickShowListView()
            val selectedItemText = clickListViewItem(Random.nextInt(getListViewCount()))
            assertSelectedItemIsDisplayed(selectedItemText)
        }
    }
}
